website/docs/ref/macro.md

```diff
@@ -241,59 +275,73 @@ By default, the `i18n` object is imported from `@lingui/core`. If you use a cust
 
 *Message Descriptor* is an object with message ID, default message and other parameters. [`i18n._`](/docs/ref/core.md#i18n._) accepts message descriptors and performs translation and formatting:
 
-``` jsx
+```ts
 type MessageDescriptor = {
-   id: String,
-   message?: String,
-   values?: Object,
-   formats?: Object,
+   id: string,
+   message?: string,
+   values?: Record<string, any>,
    comment?: string
 }
 ```
 
-`id` is message ID and the only required parameter. `id` and `message` are extracted to message catalog. Only `id`, `values`, and `formats` are used at runtime, all other attributes are removed from production code for size optimization.
+`id` is message ID and the only required parameter. `id` and `message` are extracted to message catalog. Only `id` and `values` are used at runtime, all other attributes are removed from production code for size optimization.
+
+You don't need to provide ID manually. Macro will automatically create a short id from your message.
+
+#### `context`
+
+The same text elements with different context are extracted with different IDs. For example, if the word "right" uses the following two definitions in two different locations, the word is translated differently and merged back into the application as different translation entries.
+
+- correct as in "you are right"
+- direction as in "turn right"
+
+If the same text elements meet the following conditions, the text elements are extracted only once and use the same ID.
```

Comment:
When reading the docs, this section doesn't look it belongs here – the `context` property is not metioned in the `MessageDescriptor` type above.

Reply:
Change it as you wish.